service/frontend/workflow_handler.go

```diff
@@ -537,7 +537,85 @@ func (wh *WorkflowHandler) PollWorkflowTaskQueue(ctx context.Context, request *w
 		return nil, err
 	}
 
-	return wh.pollWorkflowTaskQueue(ctx, request)
```

Comment:
Plz put the new logic behind the accessHistory config so we can do a controlled rollout. You can check other workflow history related APIs in frontend for an example.

Reply:
Do you have any suggestions about what to rename the dc to? I don't want to interfere with cells that currently have it set. I think `frontend.accessHistoryFraction` still makes some sense for matching and history because if those do not return history, frontend will still eventually do the lookup.